analyzers/src/SonarAnalyzer.Shared/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -43,5 +44,13 @@ public static bool ContainsGetOrSetOnDependencyProperty(this SyntaxNode syntaxNo
                    .Where(invocation => invocation.Expression.NameIs("GetValue") || invocation.Expression.NameIs("SetValue"))
                    .Any(invocation => semanticModel.GetSymbolInfo(invocation).Symbol.ContainingType.DerivesFrom(KnownType.System_Windows_DependencyObject));
         }
+
+        public static IEnumerable<StatementSyntax> GetPreviousStatementsCurrentBlock(this SyntaxNode expression)
+        {
+            var statement = expression.FirstAncestorOrSelf<StatementSyntax>();
```

Comment:
I think we should add a test for the uncovered branch. We can add `SyntaxNodeExtensionsTest` similar with [SyntaxTokenExtensionsTests](https://github.com/SonarSource/sonar-dotnet/blob/1711a5f773013925e079d11e32bf0a3f592a6706/analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SyntaxTokenExtensionsTests.cs).

Reply:
Tests were added